Framework/Indexing/src/SpectrumNumberTranslator.cpp

```diff
@@ -88,5 +109,21 @@ SpectrumIndexSet SpectrumNumberTranslator::makeIndexSet(
   return SpectrumIndexSet(indices, m_globalToLocal.size());
 }
 
+bool SpectrumNumberTranslator::isPartitioned() const {
+  // Careful: Do NOT use maps of spectrum numbers for size check, since those
+  // can (currently) be smaller than the actual size due to duplicate spectrum
+  // numbers.
+  return m_globalToLocal.size() != m_globalSpectrumNumbers.size();
+}
+
+void SpectrumNumberTranslator::checkUniqueSpectrumNumbers() const {
+  // To support legacy code that creates workspaces with duplicate spectrum
+  // numbers we check for bad spectrum numbers only when needed, i.e., when
+  // accessing index maps.
+  if (m_globalSpectrumNumbers.size() != m_partitions.size())
```

Comment:
Could `m_partitions` do with a better name? Something like `m_spectrumToPartionMap`? I got confused reading this because I keep thinking `m_partitions.size()` tells you the number of partitions.